<?xml version="1.0" ?>
<launch>
  <!-- args for franks initialization-->
  <arg name="robot_ip" default="172.16.0.2"/>
  <arg name="load_gripper" default="true" />

  <!-- Fake Transform to overlay world on panda_link0 -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="virtual_joint_broadcaster_1" args="0 0 0 0 0 0 world panda_link0" />

  <!-- Initialize real robot and gripper, moveit, and rviz-->
  <include file="$(find panda_moveit_config)/launch/franka_control.launch">
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <arg name="load_gripper" value="$(arg load_gripper)" />
  </include>

  <!-- Initialize my own interface nodes -->
  <rosparam command="load" file="$(find panda_tools)/config/params.yaml" />
  <node name="ArmCommander" pkg="panda_tools" type="ArmCommander.py" respawn="false" output="screen"></node>
  <node name="GripperCommander" pkg="panda_tools" type="GripperCommander.py" respawn="false" output="screen"></node>

  <!-- Calibration Data (New Calib Data) -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="hand_to_cameraTemp" args="0.07085917397 0.03292617131 0.05131155109  -0.00814520629  0.0006262179598  -0.7175730906  0.6964352971  panda_hand camera_link_temp" />

  <!-- Camera Temp to Camera Link-->
  <node pkg="tf2_ros" type="static_transform_publisher" name="cameraTemp_to_camera" args="0.015 0.001 0 0.5 -0.5 0.5 0.5  camera_link_temp camera_link" />
</launch>

